app/models/app.js

```diff
@@ -17,6 +17,7 @@ export default DS.Model.extend(ProvisionableMixin, {
   lastDeployOperation: DS.belongsTo('operation', {async:true}),
   currentImage: DS.belongsTo('image', {async:true}),
 
+  reloadWhileProvisioning: true,
```

Comment:
@sandersonet I refactored `shouldReload()` so that it looks for a property called `reloadOn`. That's just an array of statuses which are the statuses the model will reload on. This also removes the `SHOULD_RELOAD` boolean, since if we don't want to reload the model whatsoever, we just send in an empty array.

The `reloadOn` properties are currently string arrays, since when I tried to send `STATUSES` I got a bunch of undefined errors. I'm guessing this has something to do with not exporting the `STATUSES` on the models?